shell/platform/windows/flutter_windows_view.cc

```diff
@@ -25,17 +25,49 @@ constexpr std::chrono::milliseconds kWindowResizeTimeout{100};
 /// to be blocked until the frame with the right size has been rendered. It
 /// should be kept in-sync with how the engine deals with a new surface request
 /// as seen in `CreateOrUpdateSurface` in `GPUSurfaceGL`.
-bool SurfaceWillUpdate(size_t cur_width,
-                       size_t cur_height,
-                       size_t target_width,
-                       size_t target_height) {
+static bool SurfaceWillUpdate(size_t cur_width,
+                              size_t cur_height,
+                              size_t target_width,
+                              size_t target_height) {
   // TODO (https://github.com/flutter/flutter/issues/65061) : Avoid special
   // handling for zero dimensions.
   bool non_zero_target_dims = target_height > 0 && target_width > 0;
   bool not_same_size =
       (cur_height != target_height) || (cur_width != target_width);
   return non_zero_target_dims && not_same_size;
 }
+
+/// Update the surface's swap interval to block until the v-blank iff
+/// the system compositor is disabled.
+static void UpdateVsync(FlutterWindowsEngine& engine,
+                        WindowBindingHandler& window) {
+  AngleSurfaceManager* surface_manager = engine.surface_manager();
+  if (!surface_manager) {
+    return;
+  }
+
+  // Updating the vsync makes the EGL context and render surface current.
+  // If the engine is running, the render surface should only be made current on
+  // the raster thread. If the engine is initializing, the raster thread doesn't
+  // exist yet and the render surface can be made current on the platform
+  // thread.
```

Comment:
Is this operation safe/effective on either the raster or platform thread? Or in other words, should we expect it to work both before and after the engine initializes? And if so, what is the reasoning for posting it as a task only when it is possible to do so?

Reply:
> Is this operation safe/effective on either the raster or platform thread? 

No. The EGL context cannot be current on multiple threads. Since the raster thread uses the EGL context without any synchronization, other threads shouldn't use the EGL context at all while the raster thread exists. In other words, this operation must not run on the platform thread while the engine is running and the raster thread exists.

However, the EGL context is unused before the engine is initialized. Thus, this operation is safe on the platform thread during startup.

> should we expect it to work both before and after the engine initializes?

Yup

Reply:
What is achieved by running it on the platform thread when the engine is not initialized? Do we want to make the EGL context current at all on the platform thread in the case where there is not yet any raster thread?

Reply:
Our end goal is to set the surface's swap interval to `0`. Without this, you'd see Windows takes ~17ms - the full frame budget - to render each surface as EGL's swap buffer blocks until the v-blank. This would prevent us from rendering multiple views using a single raster thread on Windows.

To update the swap interval, you first must first make the EGL context current.

Ideally, only the raster thread would make the EGL context current. However, doing so would require us to update the swap intervals in 3 locations:

1. When a view is created and the engine is running, we can update its surface's swap interval by posting to the raster thread
2. For views were created before the engine was started, these must be updated once the engine starts 
3. When the system compositor changes, we need to update surfaces' swap intervals (the blocking until v-blank prevents screen tearing in this scenario)

Using the platform thread allows us to remove that 2nd location by allowing us to update the swap interval whenever a view is created.